ToolBox/LeanDataWriter.cs

```diff
@@ -87,6 +100,229 @@ public void Write(IEnumerable<BaseData> source)
             }
         }
 
+        /// <summary>
+        /// Downloads historical data from the brokerage and saves it in LEAN format.
+        /// </summary>
+        /// <param name="brokerage">The brokerage from where to fetch the data</param>
+        /// <param name="symbols">The list of symbols</param>
+        /// <param name="startTimeUtc">The starting date/time (UTC)</param>
+        /// <param name="endTimeUtc">The ending date/time (UTC)</param>
+        public void DownloadAndSave(IBrokerage brokerage, List<Symbol> symbols, DateTime startTimeUtc, DateTime endTimeUtc)
+        {
+            if (symbols.Count == 0)
+            {
+                throw new Exception("The symbol list cannot be empty.");
+            }
+
+            if (_tickType != TickType.Trade && _tickType != TickType.Quote)
+            {
+                throw new Exception("The tick type must be Trade or Quote.");
+            }
+
+            var dataType = LeanData.GetDataType(_resolution, _tickType);
+
+            var marketHoursDatabase = MarketHoursDatabase.FromDataFolder();
+
+            var ticker = symbols.First().ID.Symbol;
+            var market = symbols.First().ID.Market;
+
+            var canonicalSymbol = Symbol.CreateFuture(ticker, market, SecurityIdentifier.DefaultDate);
```

Comment:
Maybe at the start we should assert all the provided symbols correspond to the same Future, else this might get confusing/messy. Also we should assert the security type is a Future? what about options?
Worth adding to the xml method documentation too.